### ISSUE TITLE
Event callbacks thread

### ISSUE THREAD

**Issue by TheChocolateOre** [glfw/glfw#2119]
Will event callbacks be called **only** from the **main thread**?

From the `glfwPollEvents` doc:
> Thread safety: This function must only be called from the main thread.

Of course that does not imply that callbacks will be called from the same **main** thread, but it's a step in that direction I guess.

> Do not assume that callbacks you set will only be called in response to event processing functions like this one.

Of course that does not imply that callbacks will be called from a **different** thread, but it's a step in that direction I guess.

There isn't a contradiction about threads in the doc, but it's not clear either.
I need to know if callbacks will be called from different threads than the main, to make them thread-safe.

**Comment by elmindreda** [comment]
Good point. It is stated explicitly [here](https://www.glfw.org/docs/latest/intro#thread_safety), but it should be said in the reference docs of each event processing function as well. Thank you!
